packages/fxa-payments-server/src/routes/Checkout/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useContext, useMemo } from 'react';
+import { connect } from 'react-redux';
+import { AppContext } from '../../lib/AppContext';
+import { LoadingOverlay } from '../../components/LoadingOverlay';
+import { PlanErrorDialog } from '../../components/PlanErrorDialog';
+import { useMatchMedia } from '../../lib/hooks';
+import { getSelectedPlan } from '../../lib/plan';
+
+import { State } from '../../store/state';
+import { sequences, SequenceFunctions } from '../../store/sequences';
+import { selectors, SelectorReturns } from '../../store/selectors';
+
+import '../Checkout/index.scss';
+
+export type CheckoutProps = {
+  match: {
+    params: {
+      productId: string;
+    };
+  };
+  plans: SelectorReturns['plans'];
+  plansByProductId: SelectorReturns['plansByProductId'];
+  fetchCheckoutRouteResources: SequenceFunctions['fetchCheckoutRouteResources'];
+};
+
+export const Checkout = ({
+  match: {
+    params: { productId },
+  },
+  plans,
+  plansByProductId,
+  fetchCheckoutRouteResources,
+}: CheckoutProps) => {
+  const { locationReload, queryParams, matchMediaDefault } =
+    useContext(AppContext);
+
+  const isMobile = !useMatchMedia('(min-width: 768px)', matchMediaDefault);
+  const planId = queryParams.plan;
+
+  // Fetch plans on initial render or change in product ID
+  useEffect(() => {
+    fetchCheckoutRouteResources();
+  }, [fetchCheckoutRouteResources]);
+
+  const selectedPlan = useMemo(
+    () => getSelectedPlan(productId, planId, plansByProductId),
+    [plans]
+  );
+
+  if (plans.loading) {
+    return <LoadingOverlay isLoading={true} />;
+  }
+
+  if (!plans.result || plans.error !== null || !selectedPlan) {
```

Comment:
In Product and Checkout routes, I combined the `if` blocks for plan errors into one and conditionally load a new component, `PlanErrorDialog` to try making things DRYer, but I’m still repeating the `if` conditions in both routes (i.e. `if (!plans.result || plans.error !== null || !selectedPlan) {...}`). Is there a DRYer way to do this?

Reply:
This makes sense to me, I feel it is DRY enough for these cases, while still being clear to the next engineer in the file.